llrt_core/src/modules/llrt/uuid.rs

```diff
@@ -60,6 +62,64 @@ pub fn uuidv4() -> String {
     Uuid::new_v4().format_hyphenated().to_string()
 }
 
+fn uuidv6() -> String {
+    Uuid::now_v6(&NODE_ID).format_hyphenated().to_string()
+}
+
+fn uuidv7() -> String {
+    Uuid::now_v7().format_hyphenated().to_string()
+}
+
+fn uuidv1_to_v6<'js>(ctx: Ctx<'js>, v1_value: Value<'js>) -> Result<String> {
+    let v1_uuid = from_value(&ctx, v1_value)?;
+    let v1_bytes = v1_uuid.as_bytes();
+    let mut v6_bytes = [0u8; 16];
+
+    // time_high
+    v6_bytes[0] = ((v1_bytes[6] & 0x0f) << 4) | ((v1_bytes[7] & 0xf0) >> 4);
```

Comment:
Note: the Node.js package implementation does `((v1_bytes[7] >> 4) & 0x0f)` instead for getting the upper 4 bits of the 7th byte, which is used to set the lower 4 bits of the 0th byte of the uuidv6.

_I think_ there's no differences, plus the rest of the lines does not do that.

I made a comment on their PR to ask about this: https://github.com/uuidjs/uuid/pull/754#discussion_r1703097117